### PR TITLE
feat(step_7): add plain diff format

### PR DESCRIPTION
**Pull request by AlexMomot-717**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,4 +8,4 @@
 
 ### Demo
 
-[![asciicast](https://asciinema.org/a/612701.svg)](https://asciinema.org/a/612701)
+[![asciicast](https://asciinema.org/a/612870.svg)](https://asciinema.org/a/612870)
```
Comment (svntmr) [inline README.md]: you should add new asciinema instead of replacing the old one :)

File: gendiff/core.py [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 ) -> str:
     f1 = parse_file(first_file)
     f2 = parse_file(second_file)
-    if output_format == "stylish":
+    if format_name == "stylish":
         return get_diff_str_nested(f1, f2)
+    elif format_name == "plain":
+        return decorate_plain_diff(f1, f2)
     else:
         return get_diff_str(f1, f2)
```
Comment (svntmr) [inline gendiff/core.py]: time to remove this code

File: gendiff/formats/plain.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+from typing import Any, Dict
+
+from gendiff.diff import get_diff_view
+
+
+def convert_value_plain_format(val: Any) -> Any:
+    exceptions_to_convert = {False: "false", True: "true", None: "null"}
+    if val in exceptions_to_convert.keys():
```
Comment (svntmr) [inline gendiff/formats/plain.py]: `val in exceptions_to_convert` will do the same

File: gendiff/formats/plain.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+from typing import Any, Dict
+
+from gendiff.diff import get_diff_view
+
+
+def convert_value_plain_format(val: Any) -> Any:
```
Comment (svntmr) [inline gendiff/formats/plain.py:12]: this function returns string

File: gendiff/formats/plain.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            value = "[complex value]"
+        else:
+            value = convert_value_plain_format(value)
+        if isinstance(ch_value, dict):
+            ch_value = "[complex value]"
+        else:
+            ch_value = convert_value_plain_format(ch_value)
+
+        if key_type in ("added", "added_nested"):
+            diff_str += f"Property '{prefix}{key}' was added with value: {value}\n"
```
Comment (svntmr) [inline gendiff/formats/plain.py]: let's add templates as in `stylish`

File: tests/test_core.py [hunk 6]
```diff
@@ -49,10 +53,10 @@ def test_generate_diff_miss_format() -> None:
     # given
     file1 = str(resolve_path("fixtures/file1.json"))
     file2 = str(resolve_path("fixtures/file2.json"))
-    output_format = "simple"
+    format_name = "flat"
```
Comment (svntmr) [inline tests/test_core.py]: should be replaced to `stylish` (as `flat` code will be removed)

**Review by svntmr — CHANGES_REQUESTED** [review]
almost!

File: gendiff/formats/plain.py (hunk 3 above)
Comment (AlexMomot-717) [reply gendiff/formats/plain.py]: special thanks for the comment))
I've thought more and done really better)

File: gendiff/formats/plain.py (hunk 4 above)
Comment (AlexMomot-717) [reply gendiff/formats/plain.py:12]: now it returns Any.
if declire returning str, mypy fails with:

run mypy.................................................................Failed
- hook id: mypy
- exit code: 1
gendiff/formats/plain.py:18: error: Returning Any from function declared to return "str"  [no-any-return]

Comment (svntmr) [reply gendiff/formats/plain.py:12]: try to declare `exceptions_to_convert: Dict[bool | None, str]`

File: tests/test_core.py [hunk 7]
```diff
@@ -35,10 +39,10 @@ def test_generate_diff_mixed() -> None:
     # given
     file1 = str(resolve_path("fixtures/file1.yaml"))
     file2 = str(resolve_path("fixtures/file2.json"))
-    output_format = "simple"
+    format_name = "stylish"
```
Comment (svntmr) [inline tests/test_core.py]: `stylish` is value by default, consider removing it :)

**Review by svntmr — CHANGES_REQUESTED** [review]

**Review by svntmr — APPROVED** [review]

**Pull request merged**